core/metrics-core-impl/src/main/java/org/hawkular/metrics/core/impl/MetricsServiceImpl.java

```diff
@@ -103,7 +100,7 @@
         private final MetricId metricId;
 
         public DataRetentionKey(String tenantId, MetricType type) {
-            metricId = new MetricId(tenantId, type, "[" + type.getText() + "]");
+            metricId = new MetricId(tenantId, type, "$" + type.getText());
```

Comment:
Yes it should. And we probably need a separate ticket along with updated documentation to formally introduce reserved characters.